console_backend/src/utils.rs

```diff
@@ -16,6 +17,46 @@ use crate::errors::*;
 use crate::shared_state::{ConnectionState, SerialConfig, SharedState};
 use crate::types::SignalCodes;
 
+pub fn app_dir() -> crate::types::Result<PathBuf> {
+    std::env::current_exe()?
+        .parent()
+        .ok_or(anyhow::format_err!("no parent directory"))
+        .map(Path::to_path_buf)
+}
+
+/// Returns directory to packaged python, or workspace when ran locally in dev environment
+/// This is used to locate rtcm3tosbp
+pub fn pythonhome_dir() -> crate::types::Result<PathBuf> {
```

Comment:
i definitely appreciate all the comments, which helps a lot but this feels a little hacky. Unfortunately I don't have a better solution off the top of my head. I would consider making `"py39"` a global const though, and try to link it to where-ever it is comes from when created the dev env folder

Reply:
yeah this is really hacky, think the better way might be passing things via maketoml
but this should be ok since swift console executable must be within these folders.
*also didn't want to spend a bunch of time tinkering the maketoml, iirc a bunch of hardcoded py39 paths there when trying to upgrade to py311...